modules/bluefish/consumer/bluefish_consumer.cpp

```diff
@@ -277,32 +475,31 @@ struct bluefish_consumer : boost::noncopyable
 						static_cast<int>(frame.audio_data().size()/channel_layout_.num_channels), 
 						static_cast<int>(channel_layout_.num_channels));
 								
-			blue_->system_buffer_write_async(const_cast<uint8_t*>(reserved_frames_.front()->image_data()), 
-											static_cast<unsigned long>(reserved_frames_.front()->image_size()), 
-											nullptr, 
-											BlueImage_HANC_DMABuffer(reserved_frames_.front()->id(), BLUE_DATA_IMAGE));
+			blue_->system_buffer_write(const_cast<uint8_t*>(reserved_frames_.front()->image_data()), 
+											static_cast<unsigned long>(reserved_frames_.front()->image_size()),  
+											BlueImage_HANC_DMABuffer(reserved_frames_.front()->id(), BLUE_DATA_IMAGE), 
+											0);
 
-			blue_->system_buffer_write_async(reserved_frames_.front()->hanc_data(),
-											static_cast<unsigned long>(reserved_frames_.front()->hanc_size()), 
-											nullptr,                 
-											BlueImage_HANC_DMABuffer(reserved_frames_.front()->id(), BLUE_DATA_HANC));
+			blue_->system_buffer_write(reserved_frames_.front()->hanc_data(),
+											static_cast<unsigned long>(reserved_frames_.front()->hanc_size()),
+											BlueImage_HANC_DMABuffer(reserved_frames_.front()->id(), BLUE_DATA_HANC),
+											0);
 
 			if(BLUE_FAIL(blue_->render_buffer_update(BlueBuffer_Image_HANC(reserved_frames_.front()->id()))))
 				CASPAR_LOG(warning) << print() << TEXT(" render_buffer_update failed.");
 		}
 		else
 		{
-			blue_->system_buffer_write_async(const_cast<uint8_t*>(reserved_frames_.front()->image_data()),
+			blue_->system_buffer_write(const_cast<uint8_t*>(reserved_frames_.front()->image_data()),
 											static_cast<unsigned long>(reserved_frames_.front()->image_size()), 
-											nullptr,                 
-											BlueImage_DMABuffer(reserved_frames_.front()->id(), BLUE_DATA_IMAGE));
+											BlueImage_DMABuffer(reserved_frames_.front()->id(), BLUE_DATA_IMAGE),
+											0);
```

Comment:
It seems to be this `bfcSystemBufferWriteAsync()` call that takes almost linearly more time when more SDI outputs are used at the same time on the same card. So when I run 3 simultaneous channels this call takes 3 times as long on each channel as it does when only using one channel. They seem to be contending for some resource.

What does the `OVERLAPPED* pOverlap` parameter do? It is always set to `nullptr`. Does that mean that it is not asynchronous but blocking?